src/main/java/com/exactpro/epfast/template/assertion/InstrEqualityChecker.java

```diff
@@ -0,0 +1,160 @@
+        return areSameReferences(actual.getTemplateRef(), instr.getTemplateRef());
+    }
+
+    private static boolean int32Equality(Int32Field actual, Int32Field instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean int64Equality(Int64Field actual, Int64Field instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean uInt32Equality(UInt32Field actual, UInt32Field instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean uInt64Equality(UInt64Field actual, UInt64Field instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean asciiStringEquality(AsciiStringField actual, AsciiStringField instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean unicodeStringEquality(UnicodeStringField actual, UnicodeStringField instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator()) &&
+            areSameIdentities(actual.getLengthFieldId(), instr.getLengthFieldId());
+    }
+
+    private static boolean byteVectorEquality(ByteVectorField actual, ByteVectorField instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator()) &&
+            areSameIdentities(actual.getLengthFieldId(), instr.getLengthFieldId());
+    }
+
+    private static boolean compoundDecEquality(CompoundDecimalField actual, CompoundDecimalField instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getExponent(), instr.getExponent()) &&
+            areSameOperators(actual.getMantissa(), instr.getMantissa());
+    }
+
+    private static boolean simpleDecEquality(SimpleDecimalField actual, SimpleDecimalField instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean groupEquality(Group actual, Group instr) {
+        return fieldInstrEquality(actual, instr) && areSameReferences(actual.getTypeRef(), instr.getTypeRef()) &&
+            areSameInstructions(actual.getInstructions(), instr.getInstructions());
+    }
+
+    private static boolean sequenceEquality(Sequence actual, Sequence instr) {
+        return fieldInstrEquality(actual, instr) && areSameReferences(actual.getTypeRef(), instr.getTypeRef()) &&
+            areSameInstructions(actual.getInstructions(), instr.getInstructions()) &&
+            (Objects.equals(actual.getLength(), instr.getLength()) ||
```

Comment:
`Objects.equals` calls `.equal()` method inside. It's not that you need here. Here `==` should be used.